runtime/lua/vim/lsp/completion.lua

```diff
@@ -540,7 +540,7 @@ local function on_complete_done()
     -- Remove the already inserted word.
     local start_char = cursor_col - #completed_item.word
     local line = api.nvim_buf_get_lines(bufnr, cursor_row, cursor_row + 1, true)[1]
-    api.nvim_buf_set_text(bufnr, cursor_row, start_char, cursor_row, #line, { '' })
+    api.nvim_buf_set_text(bufnr, cursor_row, start_char, cursor_col, #line, { '' })
```

Comment:
If this "works", I would guess it's only by accident. The parameter is a row, not a column.

The `#line` arg looks suspicious, is that intentional @MariaSolOs ?

Reply:
I agree that changing the `row` parameter to `col` might just be making the tests pass but not be a correct change.

Regarding `#line`, what do you find suspicious about it @justinmk?

Reply:
Ah, I see. Being focused on creating the correct test I left the wrong untested change. The original change I tested manually was placed inside installed lua files and lost with next `make install`. Without checking I see that `#line` (not `cursor_row` should be probably change to `cursor_col`).

Reply:
Even after changing the argument there is an incorrect "suffix" that is pasted after the completed word.

Reply:
I see that returning empty string can avoid that but not aware possibly side effect (and all tests from `./test/functional/plugin/lsp/completion_spec.lua` passed for me).

Reply:
> Even after changing the argument there is an incorrect "suffix" that is pasted after the completed word.

You'd also have to adapt the `suffix` logic.
`apply_snippet` no longer needs to have a `suffix` parameter and `clear_word` doesn't need to have a return value

```diff
diff --git a/runtime/lua/vim/lsp/completion.lua b/runtime/lua/vim/lsp/completion.lua
index 71ea2df10..fa9cf3e88 100644
--- a/runtime/lua/vim/lsp/completion.lua
+++ b/runtime/lua/vim/lsp/completion.lua
@@ -111,16 +111,15 @@ local function parse_snippet(input)
   end)
   return ok and tostring(parsed) or input
 end
 
 --- @param item lsp.CompletionItem
---- @param suffix? string
-local function apply_snippet(item, suffix)
+local function apply_snippet(item)
   if item.textEdit then
-    vim.snippet.expand(item.textEdit.newText .. suffix)
+    vim.snippet.expand(item.textEdit.newText)
   elseif item.insertText then
-    vim.snippet.expand(item.insertText .. suffix)
+    vim.snippet.expand(item.insertText)
   end
 end
 
 --- Returns text that should be inserted when a selecting completion item. The
 --- precedence is as follows: textEdit.newText > insertText > label
@@ -537,19 +536,16 @@ local function on_complete_done()
       return nil
     end
 
     -- Remove the already inserted word.
     local start_char = cursor_col - #completed_item.word
-    local line = api.nvim_buf_get_lines(bufnr, cursor_row, cursor_row + 1, true)[1]
-    api.nvim_buf_set_text(bufnr, cursor_row, start_char, cursor_row, #line, { '' })
-    return line:sub(cursor_col + 1)
+    api.nvim_buf_set_text(bufnr, cursor_row, start_char, cursor_row, cursor_col, { '' })
   end
 
-  --- @param suffix? string
-  local function apply_snippet_and_command(suffix)
+  local function apply_snippet_and_command()
     if expand_snippet then
-      apply_snippet(completion_item, suffix)
+      apply_snippet(completion_item)
     end
 
     local command = completion_item.command
     if command then
       client:_exec_cmd(command, { bufnr = bufnr }, nil, function()
@@ -563,37 +559,37 @@ local function on_complete_done()
       end)
     end
   end
 
   if completion_item.additionalTextEdits and next(completion_item.additionalTextEdits) then
-    local suffix = clear_word()
+    clear_word()
     lsp.util.apply_text_edits(completion_item.additionalTextEdits, bufnr, offset_encoding)
-    apply_snippet_and_command(suffix)
+    apply_snippet_and_command()
   elseif resolve_provider and type(completion_item) == 'table' then
     local changedtick = vim.b[bufnr].changedtick
 
     --- @param result lsp.CompletionItem
     client.request(ms.completionItem_resolve, completion_item, function(err, result)
       if changedtick ~= vim.b[bufnr].changedtick then
         return
       end
 
-      local suffix = clear_word()
+      clear_word()
       if err then
         vim.notify_once(err.message, vim.log.levels.WARN)
       elseif result and result.additionalTextEdits then
         lsp.util.apply_text_edits(result.additionalTextEdits, bufnr, offset_encoding)
         if result.command then
           completion_item.command = result.command
         end
       end
 
-      apply_snippet_and_command(suffix)
+      apply_snippet_and_command()
     end, bufnr)
   else
-    local suffix = clear_word()
-    apply_snippet_and_command(suffix)
+    clear_word()
+    apply_snippet_and_command()
   end
 end
 
 --- @class vim.lsp.completion.BufferOpts
 --- @field autotrigger? boolean Whether to trigger completion automatically. Default: false
```

Reply:
> Regarding `#line`, what do you find suspicious about it @justinmk?

Like in #30656 - the part from the end of the word to the end of line is appended to the string that is then expanded as a snippet.